src/Microsoft.AspNet.Mvc.DataAnnotations/DataTypeAttributeAdapter.cs

```diff
@@ -17,9 +16,14 @@ public class DataTypeAttributeAdapter : DataAnnotationsClientModelValidator<Data
     {
         public DataTypeAttributeAdapter(
             DataTypeAttribute attribute,
-            [NotNull] string ruleName)
+            string ruleName)
             : base(attribute)
         {
+            if (ruleName == null)
+            {
+                throw new ArgumentNullException(nameof(ruleName));
+            }
+
             if (string.IsNullOrEmpty(ruleName))
```

Comment:
Do we want a different exception type and message for `null` and `string.Empty`? Elsewhere we seem to _just_ check `string.IsNullOrEmpty()`.